tests/e2e/utils/helpers.js

```diff
@@ -0,0 +1,70 @@
+const axios = require('axios')
+const https = require('https')
+const fs = require('fs')
+const path = require('path')
+
+const { config } = require('../cucumber.conf')
+
+const baseUrlOcis = config.baseUrlOcis
+
+const userDetails = []
+
+const makeApiRequest = async (method, url, headers, data) => {
+  try {
+    await axios({
+      method,
+      url,
+      headers,
+      data,
+      httpsAgent: new https.Agent({ rejectUnauthorized: false })
+    })
+  } catch (error) {
+    const errorMessage = `API request failed: ${method} ${url} - Headers: ${JSON.stringify(
+      headers
+    )} - Data: ${JSON.stringify(data)} - Error: ${error.message}`
+    throw new Error(errorMessage)
+  }
+}
+
+const getUserCredentials = (user) => {
+  if (user === 'admin') {
+    return { username: config.adminUser, password: config.adminPassword }
+  } else {
+    throw new Error('user not found!')
+  }
+}
+
+const getAuthHeaders = (user) => {
+  try {
+    const { username, password } = getUserCredentials(user)
+    const headers = {
+      Authorization: `Basic ${Buffer.from(`${username}:${password}`).toString('base64')}`,
+      'Content-Type': 'text/markdown'
+    }
+    return headers
+  } catch (error) {
+    throw new Error(error.message)
+  }
+}
+
+const uploadFileToServer = async (fileName, user) => {
+  const fileUploadUrl = `${baseUrlOcis}/dav/files/${user}/${fileName}`
```

Comment:
use function to build urls
```suggestion
  const fileUploadUrl = getWebDavUrl(user, fileName)
```